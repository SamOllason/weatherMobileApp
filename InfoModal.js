import React from 'react';
import {View, Text, Modal, Alert, StyleSheet, Platform, Linking, Button} from 'react-native';
import BackgroundImage from './BackgroundImage';

const InfoModal = ({modalVisible, closeModal}) => {
    const platform = Platform.select({
        ios: 'iOS',
        android: 'Android',
    });

    const projectLink =
        <Text style={{color: 'blue'}}
              onPress={() => Linking.openURL('https://github.com/SamOllason/mobileWeatherApp')}>
            here
        </Text>;

    const openWeatherMapLink =
        <Text style={{color: 'blue'}}
              onPress={() => Linking.openURL('https://openweathermap.org/api')}>
            OpenWeatherMap
        </Text>;

    return(
        <View>
            <Modal
                animationType="slide"
                transparent={false}
                visible={modalVisible}
                onRequestClose={() => {
                    Alert.alert('Modal has been closed.');
                }}>
                <BackgroundImage
                    currentWeather="clear"
                />
                <View style={styles.cont}>
                    <View style={styles.titleContainer}>
                        <Text style={{fontSize: 20}}>About</Text>
                    </View>
                    <View style={styles.textContainer}>
                        <Text>This is a cross-platform weather app developed with React Native.
                            You are currently using the {platform} version of the app.
                        </Text>
                    </View>
                    <View style={styles.textContainer}>
                        <Text>Functionality: you can enter the name of a city and see the current weather
                            and conditions in that city. You will also see a 5-day/3-hour forecast for
                            the city's weather conditions.
                        </Text>
                    </View>
                    <View style={styles.textContainer}>
                        <Text>Data: this data is provided by {openWeatherMapLink}. There is currently limit
                            on of 60 requests per minute (across all users!) so please be patient.
                        </Text>
                    </View>
                    <View style={styles.textContainer}>
                        <Text>
                            This application was developed by Sam Ollason. All the code for this project
                            is open sourced and available to view {projectLink} for more information
                        </Text>
                    </View>
                    <View style={styles.textContainer}>
                        <Text>
                            For privacy and support information please see {projectLink}.
                        </Text>
                    </View>
                    <Button
                        onPress={closeModal}
                        title="Close"
                        accessibilityLabel="Close the information screen"
                    />
                </View>
            </Modal>
        </View>
    );
};

const styles = StyleSheet.create({
    cont: {
        flex: 1,
        margin: 20,
        marginTop: 100,
    },
    titleContainer: {
        alignItems: 'center',
        marginTop: 30,
        marginBottom: 30,
    },
    textContainer: {
        marginTop: 30,
        marginBottom: 30,
    }
});

export default InfoModal;
