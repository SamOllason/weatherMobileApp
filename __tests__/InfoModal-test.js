import 'react-native';
import React from 'react';
import InfoModal from '../InfoModal';
import { shallow } from 'enzyme';

// Using Jest + Enzyme
describe('<InfoModal />', () => {
    it('renders correctly', () => {
        expect(shallow(<InfoModal/>)).toMatchSnapshot();
    });
});