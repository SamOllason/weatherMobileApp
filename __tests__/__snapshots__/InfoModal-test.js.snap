// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoModal /> renders correctly 1`] = `
<View>
  <Component
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <BackgroundImage
      currentWeather="clear"
    />
    <View
      style={
        Object {
          "flex": 1,
          "margin": 20,
          "marginTop": 20,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 30,
            "marginTop": 30,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
            }
          }
        >
          About
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          This is a cross-platform weather app developed with React Native. You are currently using the 
          iOS
           version of the app.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          Functionality: you can enter the name of a city and see the current weather and conditions. You will also see a 5-day/3-hour forecast for the city's weather conditions.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          Data: weather data is provided by 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
          >
            OpenWeatherMap
          </Text>
          . There is currently a limit of 60 requests per minute (across all users!) so please be patient. Data is refreshed on the server every 10 minutes.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          This application was developed by Sam Ollason. Please see 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
          >
            here
          </Text>
           for more information.
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          Privacy: We do not store or log any of your data. For more privacy and support information please see 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
              }
            }
          >
            here
          </Text>
          .
        </Text>
      </View>
      <Button
        accessibilityLabel="Close the information popup"
        color="#841584"
        title="Close"
      />
    </View>
  </Component>
</View>
`;
